extensions/todoist/src/components/MenubarTask.tsx

```diff
@@ -0,0 +1,120 @@
+  useEffect(() => {
+    checkTodoistApp();
+  }, []);
+
+  const priority = priorities.find((p) => p.value === task.priority);
+
+  async function completeTask(task: Task) {
+    await showHUD("⏰ Completing task");
+    try {
+      await todoist.closeTask(task.id);
+      await showHUD("Task completed 🙌");
+      mutateTasks();
+    } catch (error) {
+      showHUD("Unable to complete task ❌");
+    }
+  }
+
+  async function changeDueDate(task: Task, dueString: string) {
+    await showHUD("⏰ Updating task due date");
+
+    try {
+      await todoist.updateTask(task.id, { dueString });
+      await showHUD("Task Due Date Updated 🙌");
+      mutateTasks();
+    } catch (error) {
+      console.log(error);
+      showHUD("Unable to update task due date ❌");
+    }
+  }
+
+  async function changePriority(task: Task, priority: number) {
+    await showHUD("⏰ Updating task priority");
+
+    try {
+      await todoist.updateTask(task.id, { priority });
+      await showHUD("Task Priority Updated 🙌");
+      mutateTasks();
+    } catch (error) {
+      console.log(error);
+      showHUD("Unable to update task priority ❌");
+    }
+  }
+
+  async function deleteTask(task: Task) {
+    if (
+      await confirmAlert({
+        title: "Delete Task",
+        message: "Are you sure you want to delete this task?",
+        icon: { source: Icon.Trash, tintColor: Color.Red },
+      })
+    ) {
+      await showHUD("⏰ Deleting task");
+
+      try {
+        await todoist.deleteTask(task.id);
+        await showHUD("Task deleted 🙌");
+
+        mutateTasks();
+      } catch (error) {
+        console.log(error);
+        showHUD("Unable to delete task ❌");
+      }
+    }
+  }
+
+  return (
+    <MenuBarExtra.Submenu
+      title={task.content}
+      icon={priority && priority.value === 1 ? Icon.Circle : { source: Icon.Circle, tintColor: priority?.color }}
+    >
+      <MenuBarExtra.Item title="Complete Task" onAction={() => completeTask(task)} icon={Icon.Checkmark} />
+      <MenuBarExtra.Item
+        title="Open in Todoist"
+        onAction={() => {
+          isTodoistInstalled ? open(`todoist://task?id=${task.id}`) : open(task.url);
+        }}
+        icon={isTodoistInstalled ? "todoist.png" : Icon.Globe}
+      />
+      <MenuBarExtra.Submenu title="Change Due Date" icon={Icon.Clock}>
+        <MenuBarExtra.Item title="Today" icon={Icon.Calendar} onAction={() => changeDueDate(task, "today")} />
+        <MenuBarExtra.Item title="Tomorrow" icon={Icon.Sunrise} onAction={() => changeDueDate(task, "tomorrow")} />
+        <MenuBarExtra.Item
+          title="Next Week"
+          icon={Icon.ArrowClockwise}
+          onAction={() => changeDueDate(task, "next week")}
+        />
+        <MenuBarExtra.Item title="Next Weekend" icon={"🌴"} onAction={() => changeDueDate(task, "next weekend")} />
+        <MenuBarExtra.Item title="No Due Date" icon={Icon.Emoji} onAction={() => changeDueDate(task, "")} />
```

Comment:
An icon like `XMarkCircle` seems more appropriate:

<img width="738" alt="CleanShot 2022-11-19 at 12 11 33@2x" src="https://user-images.githubusercontent.com/16003285/202847787-a2034c72-e3a6-4710-9788-8eb0a4feeb27.png">

Reply:
Fixed in https://github.com/raycast/extensions/pull/3597/commits/87e0f2bce86513b4e9170e89543e0ad8f284b3e9